### PR TITLE
[wip] Bool toggle implementation

### PR DESCRIPTION
**Pull request by chriseidhof**
This pull-request accompanies [SE-0199: Adding `toggle` to `Bool`](https://github.com/apple/swift-evolution/blob/master/proposals/0199-bool-toggle.md)

I don't know how to run the test suite and what the standards are for documentation. I'm happy to improve on the PR if someone with more knowledge can point things out.

### CODE REVIEW

**Comment by xwu** [comment]
@swift-ci Please smoke test

**Review by xwu — COMMENTED** [review]
You'll need to add `public` :)

Seems like it'd make sense for this to be `_transparent` too.

File: stdlib/public/core/Bool.swift [hunk 1]
```diff
@@ -294,3 +294,19 @@ extension Bool {
...
+  /// Toggles the value of the Boolean. 
+  ///
+  /// Calling this method sets the variable to `true` if it was `false`,
+  /// and sets it to `false` if it was `true`. For example:
+  ///
+  ///    var bools = [true, false]
+  ///
+  ///    bools[0].toggle()
+  ///    // bools now contains [false, false]
+  mutating func toggle() {
```
Comment (airspeedswift) [inline stdlib/public/core/Bool.swift]: Needs to be public

File: stdlib/public/core/Bool.swift [hunk 2]
```diff
@@ -294,3 +294,19 @@ extension Bool {
     return lhs ? true : try rhs()
   }
 }
+
+extension Bool {
+  @_inlineable // FIXME(sil-serialize-all)
```
Comment (airspeedswift) [inline stdlib/public/core/Bool.swift]: I think you can drop the `FIXME(sil-serialize-all)` for this one. It's truly legitimately inlineable.

**Review by airspeedswift — CHANGES_REQUESTED** [review]

**Comment by gottesmm** [comment]
@chriseidhof We have in the docs directory information about the CI and testing. For the CI commands:

https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md

For the testing:

https://github.com/apple/swift/blob/master/docs/Testing.md

If you have any comments/improvements, they are welcome as always = ).

**Comment by airspeedswift** [comment]
@swift-ci please test

**Review by airspeedswift — APPROVED** [review]

**Comment by ryanmaxwell** [comment]
why no `func toggled() -> Self`as a pair with `mutating func toggle()`?

**Pull request merged**